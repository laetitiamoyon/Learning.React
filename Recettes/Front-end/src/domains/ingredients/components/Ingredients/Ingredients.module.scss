@import '../../../../styles/styles.scss'; 

.container
{
    background: url(../../../../images/backgrounds/ingredientList.jpg) no-repeat center  / cover;
    @include marginPadding(auto, 0 24px);
    height: 100%;
    min-height: 1000px;
}

.title
{
    @include text($textH3FontSize, $primaryColor, undefined, center);

    padding-top: 82px;
    margin-bottom: 38px;
    
    text-shadow: 2px 2px 4px black;
}

.form { @include flex(center); margin-bottom: 42px; }
@media (max-width: 400px) { .form { transform: scale(0.95); }}

.input
{
  @include widthHeight(200px, 32px);
  @include borderAndRadius($defaultBorder, $defaultBorderRadius);

  text-align: center;
  
  outline: none;
}