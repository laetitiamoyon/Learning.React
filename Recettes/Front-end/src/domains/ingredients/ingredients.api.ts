import axios, { AxiosResponse, AxiosRequestConfig } from 'axios'
import { Ingredient } from './ingredients.model';

export const url : string = 'http://localhost:3001/ingredients'
export const configuration : AxiosRequestConfig = { headers : { 'Content-Type' : 'application/json' } }

export const getAllIngredients = async () : Promise<AxiosResponse<Ingredient[]>> =>
    (await axios.get(url, configuration)).data

export const addIngredient = async (ingredient : Ingredient) : Promise<AxiosResponse<Ingredient>> =>
    (await axios.post(url, ingredient, configuration)).data

export const updateIngredient = async (ingredient: Ingredient) : Promise<AxiosResponse<Ingredient>> =>
    (await axios.put(`${url}/${ingredient.id}`, ingredient, configuration)).data

export const removeIngredient = async (id : string) : Promise<AxiosResponse<Ingredient>> =>
    (await axios.delete(`${url}/${id}`, configuration)).data