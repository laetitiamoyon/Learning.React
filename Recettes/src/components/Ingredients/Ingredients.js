import React from 'react';

const Ingredients = () => <div>
    <h1>Ingr√©dients</h1>
</div>

export default Ingredients;