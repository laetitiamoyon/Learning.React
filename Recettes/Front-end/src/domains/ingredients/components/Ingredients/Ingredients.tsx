import React, { useEffect } from 'react';
import { FC, ChangeEvent, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { selectIngredients } from '../../ingredients.selector';
import { getIngredientsRequestAction } from '../../ingredients.actions';
import styles from './Ingredients.module.scss'
import Ingredient from '../Ingredient/Ingredient';
import { useInformationMessage } from '../../../recipes/hooks/useInformationMessage';

const Ingredients : FC = () => 
{
  const ingredientsState = useSelector(selectIngredients)
  const dispatch = useDispatch()
  const [filteredIngredients, setFilteredIngredients] = useState(ingredientsState.ingredients)
  const [searchTerm, setSearchTerm] = useState("")

  useEffect(() => 
  {
    setFilteredIngredients(ingredientsState.ingredients
      .filter(r => r.title.toLowerCase()
      .includes(searchTerm.toLowerCase())))
  }, [ingredientsState, searchTerm])
  
  const onSearchTermChange = (event : ChangeEvent<HTMLInputElement>) : void => setSearchTerm(event.target.value)

  useEffect(() => { dispatch(getIngredientsRequestAction()) }, [])

  useInformationMessage()

  return <div className={styles.container}>
      <h1 className={styles.title}>Gérer mes ingrédients</h1>
      <div className={styles.form}>
        <input className={styles.input}
          placeholder="Rechercher un ingrédient"
          onChange={onSearchTermChange}/>
      </div>
      <div className={styles.ingredients}>
        { filteredIngredients.map(i => <Ingredient {...i} key={i.id} 
          searchIngredientTerm={searchTerm} title={i.title} color="#f8813d"/>)}
      </div>
  </div>
};

export default Ingredients;