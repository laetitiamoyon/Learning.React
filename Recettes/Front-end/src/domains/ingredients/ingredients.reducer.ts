import { failedActionMetadata, loadedActionMetadata, loadingActionMetadata } from '../../shared/domains/Redux/redux.utils';
import { IngredientAction, IngredientActions } from './ingredients.actions';
import { IngredientState, initialIngredients } from './ingredients.state';


export const ingredientsReducer = (state: IngredientState = initialIngredients , action: IngredientActions): IngredientState =>
{
    switch (action.type)
    {
        case IngredientAction.GET_INGREDIENTS_REQUEST : return loadingActionMetadata(state)
        case IngredientAction.GET_INGREDIENTS_SUCCESS : return {
            ...loadedActionMetadata(state),
            ingredients: action.payload
        }
        case IngredientAction.GET_INGREDIENTS_FAILED : return failedActionMetadata(action.error, state)
  
        default : return state
    }
}