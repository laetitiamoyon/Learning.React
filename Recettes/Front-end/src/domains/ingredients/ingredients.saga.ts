import { call, put, takeLatest } from 'redux-saga/effects'
import { addIngredientFailedAction, AddIngredientRequestAction, addIngredientSuccessAction, editIngredientFailedAction, EditIngredientRequestAction, editIngredientSuccessAction, getIngredientsFailedAction, getIngredientsRequestAction, GetIngredientsRequestAction, getIngredientsSuccessAction, IngredientAction, removeIngredientFailedAction, RemoveIngredientRequestAction, RemoveIngredientSuccessAction } from './ingredients.actions';
import { Ingredient } from './ingredients.model';
import { addIngredient, getAllIngredients, updateIngredient, removeIngredient } from './ingredients.api';

export function* getAllIngredientsSaga()
{
   try
   {
      const ingredients : Ingredient[] = yield call(getAllIngredients);
  
      yield put(getIngredientsSuccessAction(ingredients));
   }
   catch (error : any)
   {
      yield put(getIngredientsFailedAction(error.message));
   }
}

export function* addIngredientSaga(action : AddIngredientRequestAction)
{
   try
   {
      const newIngredient : Ingredient = yield call(addIngredient, action.payload);
      
      yield put(addIngredientSuccessAction(newIngredient));
      yield put(getIngredientsRequestAction());
   }
   catch (error : any)
   {
      yield put(addIngredientFailedAction(error.message));
   }
}

export function* editIngredientSaga(action : EditIngredientRequestAction)
{
   try
   {
      const ingredient = action.payload
      const editedIngredient : Ingredient = yield call(updateIngredient, ingredient);
     
      yield put(editIngredientSuccessAction(editedIngredient));
      yield put(getIngredientsRequestAction());
   }
   catch (error : any)
   {
      yield put(editIngredientFailedAction(error.message));
   }
}

export function* removeIngredientSaga(action : RemoveIngredientRequestAction)
{
   try
   {
        const { id } = action.payload

        yield call(removeIngredient, id);
        yield put(RemoveIngredientSuccessAction(id));
        yield put(getIngredientsRequestAction());
      }
   catch (error : any)
   {
      yield put(removeIngredientFailedAction(error.message));
   }
}

export default function* ingredientSaga() : Generator {
  yield takeLatest(IngredientAction.GET_INGREDIENTS_REQUEST, getAllIngredientsSaga);
  yield takeLatest(IngredientAction.ADD_INGREDIENT_REQUEST, addIngredientSaga);
  yield takeLatest(IngredientAction.EDIT_INGREDIENT_REQUEST, editIngredientSaga);
  yield takeLatest(IngredientAction.REMOVE_INGREDIENT_REQUEST, removeIngredientSaga);
}
