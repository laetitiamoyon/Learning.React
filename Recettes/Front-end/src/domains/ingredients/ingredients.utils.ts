import { Ingredient } from "./ingredients.model";

export const filterIngredients = (ingredients : Ingredient[], searchTerm : string) : Ingredient[] =>
    ingredients.filter(r => r.title.toLowerCase().includes(searchTerm.toLowerCase()))

// TODO : tester la couche utils (__tests__/ingredients.utils.test.ts)