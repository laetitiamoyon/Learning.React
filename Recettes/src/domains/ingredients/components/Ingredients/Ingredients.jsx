import React, { useContext } from 'react';
import styles from './Ingredients.module.css'
import AddIngredient from '../AddIngredient/AddIngredient';
import Ingredient from '../Ingredient/Ingredient';
import { IngredientContext } from '../../ingredients.provider';

const Ingredients = () => 
{
    const { ingredients } = useContext(IngredientContext)

    return <>
        <h1 className={styles.title}>Gérer les ingrédients</h1>
    
        <div className={styles.ingredientContainer}>
            { ingredients && ingredients.map(i => <Ingredient {...i} key={i.id} />)}
        </div> 
        <div><AddIngredient/></div>
    </>
}

export default Ingredients;