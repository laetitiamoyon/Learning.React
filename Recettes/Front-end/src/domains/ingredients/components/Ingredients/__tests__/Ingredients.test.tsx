import {useDispatch, useSelector} from "react-redux";
import {ingredientStateMock} from "../../../ingredient.mock";
import {getTodosRequestAction} from "../../../../../../../../todo-list/Front-end/src/domains/Todos/todos.actions";
import Ingredients from "../Ingredients";
import Todos from "../../../../../../../../todo-list/Front-end/src/domains/Todos/components/Todos/Todos";
import {act, render} from "@testing-library/react";
import {getIngredientsRequestAction} from "../../../ingredients.actions";


jest.mock('react-redux')
const mockUseDispatch = useDispatch as jest.MockedFunction<typeof useDispatch>
const mockDispatch = jest.fn()


describe("Ingredients", () => {
    beforeEach(() => {
        (mockUseDispatch as jest.Mock).mockReturnValue(mockDispatch)
    })

    it("should match snapshot", async () => {
        // Given
        (useSelector as jest.Mock).mockReturnValue(ingredientStateMock)

        // When
        const {container} = render(<Ingredients/>)

        // Then
        expect(container).toMatchSnapshot()
    })

    it("should dispatch getIngredientsRequestAction when component is mounted and ingredient reducer and saga are injected",() =>
    {
        // Given
        (useSelector as jest.Mock).mockReturnValue(ingredientStateMock);

        // When
        render(<Ingredients/>)

        // Then
        expect(mockDispatch).toBeCalledWith(getIngredientsRequestAction())
    })

    it("should filter title recipes when written some letter in the search bar and when component is mounted", () =>
    {
        // Given
        const {container} = render(<Ingredients />)
        const input = container.getElementsByTagName('input')[0]

        // When
       //fireEvent.change(input, {target: {value: }})

        // Then

    })
})