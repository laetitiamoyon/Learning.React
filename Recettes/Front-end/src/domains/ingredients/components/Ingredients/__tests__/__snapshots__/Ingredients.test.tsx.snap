// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h1
      class="title"
    >
      Gérer mes ingrédients
    </h1>
    <div
      class="form"
    >
      <input
        class="input"
        placeholder="Rechercher un ingrédient"
      />
    </div>
    <div
      class="ingredients"
    >
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              P
              <span
                style="color :#f8813d"
              />
              â
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              à
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              p
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              z
              <span
                style="color :#f8813d"
              />
              z
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="pièce"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              S
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              u
              <span
                style="color :#f8813d"
              />
              c
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              T
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              m
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              M
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              z
              <span
                style="color :#f8813d"
              />
              z
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              B
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              s
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              c
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              S
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              H
              <span
                style="color :#f8813d"
              />
              u
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              d
              <span
                style="color :#f8813d"
              />
              '
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              v
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              T
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              m
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              V
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              n
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              g
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              b
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
              s
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              m
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              q
              <span
                style="color :#f8813d"
              />
              u
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              P
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              m
              <span
                style="color :#f8813d"
              />
              m
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              s
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              P
              <span
                style="color :#f8813d"
              />
              â
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              f
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              u
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
              l
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              é
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="pièce"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              C
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              è
              <span
                style="color :#f8813d"
              />
              m
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              f
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              c
              <span
                style="color :#f8813d"
              />
              h
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              O
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              u
              <span
                style="color :#f8813d"
              />
              f
              <span
                style="color :#f8813d"
              />
              s
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="pièce"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              S
              <span
                style="color :#f8813d"
              />
              u
              <span
                style="color :#f8813d"
              />
              c
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              L
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              d
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              n
              <span
                style="color :#f8813d"
              />
              s
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              P
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              m
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              s
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              n
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              C
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              s
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              C
              <span
                style="color :#f8813d"
              />
              u
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              y
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              L
              <span
                style="color :#f8813d"
              />
              a
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              d
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
               
              <span
                style="color :#f8813d"
              />
              c
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              c
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              C
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              u
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              g
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              t
              <span
                style="color :#f8813d"
              />
              e
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="ingredient"
        >
          <div
            class="title"
          >
            <div>
              <span
                style="color :#f8813d"
              />
              P
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              i
              <span
                style="color :#f8813d"
              />
              v
              <span
                style="color :#f8813d"
              />
              r
              <span
                style="color :#f8813d"
              />
              o
              <span
                style="color :#f8813d"
              />
              n
              <span
                style="color :#f8813d"
              />
            </div>
          </div>
          <input
            class="input"
            value="gr"
          />
          <button
            class="updateButton"
          >
            Mettre à jour
          </button>
          <div
            class="trashIcon"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
